src/driver/driver.cpp

```diff
@@ -205,25 +205,60 @@ void EvolutionDriver::InitializeBlockTimeSteps() {
 // \brief function that loops over all MeshBlocks and find new timestep
 
 void EvolutionDriver::SetGlobalTimeStep() {
-  // don't allow dt to grow by more than 2x
-  // consider making this configurable in the input
-  if (tm.dt < 0.1 * std::numeric_limits<Real>::max()) {
-    tm.dt *= 2.0;
-  }
-  Real big = std::numeric_limits<Real>::max();
-  for (auto const &pmb : pmesh->block_list) {
-    tm.dt = std::min(tm.dt, pmb->NewDt());
-    pmb->SetAllowedDt(big);
+  // Check if user wants to force the value
+  if (dt_force > 0.0) {
+    tm.dt = dt_force;
+  } else {
+    // Check for special first cycle value
+    if (tm.ncycle == 0) {
+      tm.dt = std::min(tm.dt, dt_init);
+    }
+    // don't allow dt to grow by more than 2x
+    // consider making this configurable in the input
+    if (tm.dt < 0.1 * std::numeric_limits<Real>::max()) {
+      tm.dt *= 2.0;
+    }
+    // Allow the meshblocks to vote
+    Real big = std::numeric_limits<Real>::max();
+    for (auto const &pmb : pmesh->block_list) {
+      tm.dt = std::min(tm.dt, pmb->NewDt());
+      pmb->SetAllowedDt(big);
```

Comment:
```suggestion
    for (auto const &pmb : pmesh->block_list) {
      tm.dt = std::min(tm.dt, pmb->NewDt());
      pmb->SetAllowedDt(std::numeric_limits<Real>::max());
```
`big` only appears once so not sure it needs a variable

Reply:
Fixed